packages/components/src/components/floor-selector/floor-selector.tsx

```diff
@@ -0,0 +1,231 @@
+            // If the floor selector is closed, all elements need to be animated back to their original position.
+            this.animateTranslateY(this.currentFloorElement, 0, 250);
+            floorElements.forEach(floorElement => {
+                if (floorElement.getAttribute('data-floor') !== this.currentFloor) {
+                    this.animateTranslateY(floorElement as HTMLElement, 0, 250);
+                }
+            });
+        }
+    }
+
+    /**
+     * Toggles the floor selector open-close.
+     */
+    private onToggleFloorSelector(): void {
+        this.floorSelectorClosed = !this.floorSelectorClosed;
+        this.animateFloorSelector();
+
+        this.floorListElement.addEventListener('transitionend', () => {
+            if (!this.floorSelectorClosed) {
+                this.scrollToSelectedFloor();
+            }
+
+            this.onScrollStyle();
+        }, { once: true });
+    }
+
+    /**
+     * Adding/removing classes to the floorlist to fade-in/out the top/bottom part of the container..
+     */
+    private onScrollStyle(): void {
+        const maxScrollTop = this.floorListElement.scrollHeight - this.floorListElement.clientHeight;
+
+        if (this.floorSelectorClosed) {
+            this.fadeFloorSelectorOnTop = false;
+            this.fadeFloorSelectorOnBottom = false;
+            return;
+        }
+
+        if (this.maxListHeight > this.floorListElement.clientHeight - 48) {
+            return;
+        }
+
+        this.fadeFloorSelectorOnTop = this.floorListElement.scrollTop > 0 ? true : false;
+        this.fadeFloorSelectorOnBottom = this.floorListElement.scrollTop === maxScrollTop ? false : true;
+    }
+
+    /**
+     * Sets the clicked floor active.
+     *
+     * @param {PointerEvent} event
+     */
+    private onSelectFloor(event: PointerEvent, floorIndex: string): void {
+        this.mapsindoors.setFloor(floorIndex);
+        this.currentFloor = this.mapsindoors.getFloor().toString();
+        this.currentFloorElement = event.currentTarget as HTMLElement;
+    }
+
+    /**
+     * Called every time the component is connected to the DOM.
+     * It resets the floorlist and the active floor when the building changes.
+     */
+    connectedCallback(): void {
+        this.mapsindoors.addListener('building_changed', () => {
+            this.floors = [];
+            const building = this.mapsindoors.getBuilding();
+            if (building) {
+                if (!this.mapsindoors.getFloor()) {
+                    return;
+                }
+                this.currentFloor = this.mapsindoors.getFloor().toString();
+
+                Object.keys(building.floors)
+                    .sort((a, b): any => (b as any) - (a as any))
+                    .forEach(floor => {
+                        building.floors[floor].index = floor;
+                        this.floors.push(building.floors[floor]);
+                    });
+            }
+        });
+    }
+
+    /**
```

Comment:
Move this a little further down in the file. I think it's good form to have the render method at the button.